include/internal/quic_cc.h

```diff
@@ -72,14 +72,14 @@ typedef struct ossl_cc_method_st {
      * |time_valid| is 1 if the |time_since_last_send| holds
      * a meaningful value, 0 otherwise.
      */
-    size_t (*get_send_allowance)(OSSL_CC_DATA *ccdata,
-                                 OSSL_TIME time_since_last_send,
-                                 int time_valid);
+    uint64_t (*get_send_allowance)(OSSL_CC_DATA *ccdata,
+                                   OSSL_TIME time_since_last_send,
+                                   int time_valid);
 
     /*
      * Returns the maximum number of bytes allowed to be in flight.
      */
-    size_t (*get_bytes_in_flight_max)(OSSL_CC_DATA *ccdata);
+    uint64_t (*get_bytes_in_flight_max)(OSSL_CC_DATA *ccdata);
```

Comment:
So, is it ok if that maximum is limited to 2^32-1?

Reply:
For bytes in _flight_, I think 2³²-1 is more than adequate.  A machine with a 32 bit size_t cannot store that many bytes in memory and it needs to in case of retransmits.

Reply:
I agree `size_t` would be acceptable here. However the ACKM also uses a 64-bit integer to count bytes in flight.

It is worth noting that the C standard only requires `size_t` to be large enough to contain an array index, not to be pointer-sized. So theoretically you could have some weird segmented memory architecture where the total amount of RAM is much larger than the maximum size of a single allocation. So the reasoning about storing bytes for retransmission isn't strictly valid, even if it's unlikely that we'll ever support such an architecture.

I think I'd leave this as uint64_t for consistency with ACKM.

Reply:
> I think I'd leave this as uint64_t for consistency with ACKM.

Cool